lib/whitehall_importer/integrity_checker.rb

```diff
@@ -86,6 +86,30 @@ def time_matches?(proposed_time, publishing_api_time)
       self.class.time_matches?(proposed_time, publishing_api_time)
     end
 
+    def state_problems
+      problems = []
+
+      unless publishing_api_content["publication_state"] == expected_state
+        problems << problem_description("our state is not equivalent to publication_state",
```

Comment:
I think it would be more helpful in the problem description to pass in what we're expecting to see in the Publishing API, rather than what the edition's state is, since states in Content Publisher and Publishing API don't match each other.  Otherwise we could be in a scenario where the description says that we're expecting a state like 'withdrawn' which doesn't exist in the Publising API.  We could do something like:
```suggestion
        problem_description("publication_state isn't as expected", expected_state, publishing_api_content["publication_state"]
```

Reply:
I used edition state instead of the expected_state as it seemed to give a more informative message on what state the edition in content-publisher is as using the expected state could mean the edition in cp is in multiple states  (draft => scheduled, submitted_for_review, draft).

I suppose it's a tradeoff and it probably does make more sense for it to be publishing-api's expected state rather than edition state. Change incoming.